src/Build/Graph/ParallelWorkSet.cs

```diff
@@ -188,8 +188,14 @@ private void ExecuteWorkItem()
         {
             if (_queue.TryDequeue(out Lazy<TResult> workItem))
             {
-                TResult _ = workItem.Value;
-                Interlocked.Decrement(ref _pendingCount);
+                try
+                {
+                    TResult _ = workItem.Value;
+                }
+                finally
+                {
+                    Interlocked.Decrement(ref _pendingCount);
```

Comment:
@ccastanedaucf I had to put the decrement in a finally block, otherwise an exception from `workItem.Value` would miss the decrement, and block the main thread forever. Probably a good idea to move this to qb as well?

Reply:
Sounds good, I also caught this today in a branch I have where I'm filling out unit tests for ParallelWorkSet

Reply:
Nice. Maybe worth considering pulling the class into a separate repo with it's own nuget package :)